yrs/src/block.rs

```diff
@@ -991,18 +998,29 @@ impl Item {
     }
 }
 
+impl PartialEq for Item {
+    fn eq(&self, other: &Self) -> bool {
+        self.try_eq(other)
+            .expect(format!("Item equality was not possible: {} vs {}", self, other).as_str())
+    }
+}
+
 #[derive(PartialEq, PartialOrd, Eq, Ord, Debug, Clone)]
 pub struct SplittableString {
     content: SmallString<[u8; 8]>,
-    utf16_len: usize,
 }
 
 impl SplittableString {
     pub fn len(&self, kind: OffsetKind) -> usize {
-        match kind {
-            OffsetKind::Bytes => self.content.len(),
-            OffsetKind::Utf16 => self.utf16_len(),
-            OffsetKind::Utf32 => self.unicode_len(),
+        let len = self.content.len();
+        if len == 1 {
+            len // quite often strings are single-letter, so we don't care about OffsetKind
```

Comment:
Micro optimization to make computing lengths for single-char strings faster (which is quite common in transaction-per-keystroke case). Problem is that it still doesn't bring enough of speedup in terms of cached `Item::len` field. We might possibly make this field to be calculated lazily.